src/components/FlatList/index.android.js

```diff
@@ -0,0 +1,74 @@
+import React, {forwardRef, useCallback, useState} from 'react';
+import {FlatList} from 'react-native';
+import PropTypes from 'prop-types';
+import {useFocusEffect} from '@react-navigation/native';
+
+const propTypes = {
+    /** Same as for FlatList */
+    onScroll: PropTypes.func,
+
+    /** Same as for FlatList */
+    onLayout: PropTypes.func,
+
+    /** Same as for FlatList */
+    // eslint-disable-next-line react/forbid-prop-types
+    maintainVisibleContentPosition: PropTypes.object,
+
+    /** Passed via forwardRef so we can access the FlatList ref */
+    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({current: PropTypes.instanceOf(FlatList)})]).isRequired,
+};
+
+const defaultProps = {
+    /** Same as for FlatList */
+    onScroll: undefined,
+
+    /** Same as for FlatList */
+    onLayout: undefined,
+
+    /** Same as for FlatList */
+    maintainVisibleContentPosition: undefined,
+};
+
+// FlatList wrapped with the freeze component will lose its scroll state when frozen (only for Android).
+// CustomFlatList saves the offset and use it for scrollToOffset() when unfrozen.
```

Comment:
Hmm this is a WIP review I had started a few days ago that I tried to cancel just now because I never got around to finishing it. Sorry for the noise! (Not sure how to remove it now...)